### PR TITLE
Add connectivity diagram to `hpc_benchmark` documentation

### PR DESCRIPTION
**Pull request by jarsi**
During the review phase of a manuscript we realized that it might be beneficial to have figures of the most important models as part of the user level documentation. This PR graphically extends the user level documentation of the `hpc_benchmark.py` file with a detailed figure of its connectivity. The figure and its description are taken (and slightly adapted) from [Senk et al (2021)](https://arxiv.org/abs/2110.02883).

During integration of the figure we realized that the copying and linking of figures requires long relative paths, this can be addressed by a future PR, @terhorstd.

I suggest @jhnnsnk as a reviewer.

### CODE REVIEW

File: pynest/examples/hpc_benchmark.py [hunk 1]
```diff
@@ -34,6 +34,21 @@
 
 This is the standard network investigated in [1]_, [2]_, [3]_.
 
+A note on connectivity
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. image:: ../../../examples/hpc_benchmark_connectivity.svg
```
Comment (terhorstd) [inline pynest/examples/hpc_benchmark.py:40]: This path is something that we need to fix across all examples. This is also different from the Potjans example, but only the filename does not seem to work. Anyway hardcoding the fixed filenames in `conf.py` feels very wrong to me, but requires more of a structural review outside this pr.
I'd say this is fine for the case at hand.

**Review by terhorstd — APPROVED** [review]
Thanks a lot for working on the content!
Spread the word, that this kind of small enhancements/fixes are highly appreciated! We need more of these!
:rocket:

**Comment by terhorstd** [comment]
@jhnnsnk, I also added you for the content side review.

File: pynest/examples/hpc_benchmark.py [hunk 2]
```diff
@@ -34,6 +34,21 @@
...
+   :width: 50 %
+   :alt: HPC Benchmark network architecture
+   :align: right
+
+Each  neuron  receives :math:`K_{in,{\\tau} E}` excitatory  connections  randomly
+drawn  from population  E  and :math:`K_{in,\\tau I}` inhibitory  connections from
+population I. Autapses are prohibited and multapses are allowed.  Each neuron
```
Comment (jougs) [inline pynest/examples/hpc_benchmark.py:47]:
```suggestion
Each neuron receives :math:`K_{in,{\\tau} E}` excitatory connections randomly
drawn from population E and :math:`K_{in,\\tau I}` inhibitory connections from
population I. Autapses are prohibited and multapses are allowed. Each neuron
```

File: pynest/examples/hpc_benchmark.py [hunk 3]
```diff
@@ -34,6 +34,21 @@
...
+receives additional input from an external stimulating device. All delays are
```
Comment (jougs) [inline pynest/examples/hpc_benchmark.py:48]:
```suggestion
receives additional input from an external stimulation device. All delays are
```

File: pynest/examples/hpc_benchmark_connectivity.svg [hunk 4]
```diff
@@ -0,0 +1,654 @@
...
+        <rect class="BoundingBox" stroke="none" fill="none" x="1101" y="962" width="59" height="149"/>
+        <path fill="rgb(0,0,0)" stroke="none" d="M 1159,1014 C 1159,982 1148,962 1128,962 1110,962 1101,975 1101,988 1101,1001 1110,1015 1128,1015 1136,1015 1141,1011 1147,1008 L 1148,1007 C 1148,1007 1149,1007 1149,1014 1149,1046 1135,1076 1112,1100 1110,1102 1109,1103 1109,1104 1109,1109 1111,1110 1114,1110 1120,1110 1159,1071 1159,1014 Z"/>
+       </g>
+      </g>
+      <g class="com.sun.star.drawing.ClosedBezierShape">
+       <g id="id53">
+        <rect class="BoundingBox" stroke="none" fill="none" x="1288" y="589" width="514" height="515"/>
+        <path fill="rgb(0,0,0)" stroke="none" d="M 1797,607 C 1800,606 1801,602 1801,600 1801,594 1796,590 1791,590 1787,590 1785,591 1784,593 L 1291,1086 C 1289,1088 1288,1091 1288,1093 1288,1099 1292,1103 1298,1103 1300,1103 1303,1102 1305,1100 L 1797,607 Z"/>
+       </g>
+      </g>
```
Comment (jougs) [inline pynest/examples/hpc_benchmark_connectivity.svg:520]:
```suggestion
```

File: pynest/examples/hpc_benchmark_connectivity.svg [hunk 5]
```diff
@@ -0,0 +1,654 @@
...
+        <rect class="BoundingBox" stroke="none" fill="none" x="13902" y="962" width="59" height="149"/>
+        <path fill="rgb(0,0,0)" stroke="none" d="M 13960,1014 C 13960,982 13949,962 13929,962 13911,962 13902,975 13902,988 13902,1001 13911,1015 13929,1015 13937,1015 13942,1011 13948,1008 L 13949,1007 C 13949,1007 13950,1007 13950,1014 13950,1046 13936,1076 13913,1100 13911,1102 13910,1103 13910,1104 13910,1109 13912,1110 13915,1110 13921,1110 13960,1071 13960,1014 Z"/>
+       </g>
+      </g>
+      <g class="com.sun.star.drawing.ClosedBezierShape">
+       <g id="id65">
+        <rect class="BoundingBox" stroke="none" fill="none" x="14089" y="589" width="514" height="515"/>
+        <path fill="rgb(0,0,0)" stroke="none" d="M 14598,607 C 14601,606 14602,602 14602,600 14602,594 14597,590 14592,590 14588,590 14586,591 14585,593 L 14092,1086 C 14090,1088 14089,1091 14089,1093 14089,1099 14093,1103 14099,1103 14101,1103 14104,1102 14106,1100 L 14598,607 Z"/>
+       </g>
+      </g>
```
Comment (jougs) [inline pynest/examples/hpc_benchmark_connectivity.svg:600]:
```suggestion
```

**Review by jougs — COMMENTED** [review]
Just some minor suggestions from my side. The ones on the SVG file remove the diagonal lines through the "A"s next to the feedback connections, as they are not really explained in this context.

File: pynest/examples/hpc_benchmark_connectivity.svg (hunk 5 above)
Comment (jarsi) [reply pynest/examples/hpc_benchmark_connectivity.svg:600]: Thanks @jougs. The diagonal line through the A means that autapses are prohibited which is stated in the description. If needed I can make this more explicit of course.

File: pynest/examples/hpc_benchmark.py [hunk 6]
```diff
@@ -34,6 +34,21 @@
...
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. image:: ../../../examples/hpc_benchmark_connectivity.svg
+   :width: 50 %
+   :alt: HPC Benchmark network architecture
+   :align: right
+
+Each neuron receives :math:`K_{in,{\\tau} E}` excitatory connections randomly
+drawn from population E and :math:`K_{in,\\tau I}` inhibitory connections from
+population I. Autapses are prohibited and multapses are allowed. Each neuron
```
Comment (jougs) [inline pynest/examples/hpc_benchmark.py:47]:
```suggestion
population I. Autapses are prohibited (denoted by the crossed out A next to
the connections) while multapses are allowed (denoted by the M). Each neuron
```

File: pynest/examples/hpc_benchmark_connectivity.svg (hunk 5 above)
Comment (jougs) [reply pynest/examples/hpc_benchmark_connectivity.svg:600]: Ah! Thanks for clarifying. I've added a suggestion to the figure caption that would have made things clearer for at least me. Maybe it also helps others?

**Review by jougs — CHANGES_REQUESTED** [review]

File: pynest/examples/hpc_benchmark_connectivity.svg (hunk 5 above)
Comment (jarsi) [reply pynest/examples/hpc_benchmark_connectivity.svg:600]: I agree, now it's way better :)

**Review by jougs — APPROVED** [review]

**Pull request merged**